lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -298,6 +297,31 @@ public void onAuthUpdated() {
 		}
 	}
 
+	/**
+	 * Called when where was an error during authentication attempt
+	 *
+	 * @param errorInfo Error associated with unsuccessful authentication
+	 */
+	public void onAuthError(ErrorInfo errorInfo) {
+		Log.i(TAG, String.format("onAuthError: (%d) %s", errorInfo.code, errorInfo.message));
+		switch (state.state) {
+			case connecting:
+				ITransport transport = this.transport;
+				if (transport != null)
+					/* onTransportUnavailable will send state change event and set transport to null */
+					onTransportUnavailable(transport, null, errorInfo);
```

Comment:
@paddy Are you questioning whether the spec is right here? Ignoring that question, I think Peter's change implements the spec as written. However there is the issue mentioned by @matt (and I think discussed elsewhere) that the connected case does a connected->connected state change as a way of staying in connected state but emitting the error. RTN4g says there should be a way to emit an error event, but that does not exist in the java lib.